### PR TITLE
Unifiy zlib linkage by compiling it as static library and linking it to concerned projects

### PR DESCRIPTION
**Pull request by Arkshine**
It bothered me zlib was duplicated and used differently instead of having one source.
For consistency and maintainability, latest zlib is now moved to /public and AMBuild will compile it as static library and will link to concerned projects. 
Note: For VS, it will compile from source..

### CODE REVIEW

File: amxmodx/util.cpp [hunk 1]
```diff
@@ -291,6 +291,7 @@ void UTIL_ClientPrint(edict_t *pEntity, int msg_dest, char *msg)
 		MESSAGE_BEGIN(MSG_BROADCAST, gmsgTextMsg);
 	
 	WRITE_BYTE(msg_dest);
+	WRITE_STRING("%s");
```
Comment (xPaw) [inline amxmodx/util.cpp]: Is this change supposed to be part of this pull request?

File: amxmodx/util.cpp [hunk 2]
```diff
@@ -306,6 +307,7 @@ void UTIL_ClientSayText(edict_t *pEntity, int sender, char *msg)
 
 	MESSAGE_BEGIN(MSG_ONE, gmsgSayText, NULL, pEntity);
 	WRITE_BYTE(sender);
+	WRITE_STRING("%s");
```
Comment (xPaw) [inline amxmodx/util.cpp]: Is this change supposed to be part of this pull request?

File: dlls/cstrike/cstrike/msvc12/cstrike.vcxproj [hunk 3]
```diff
@@ -146,6 +146,7 @@
     <ClCompile Include="..\..\..\..\public\memtools\MemoryUtils.cpp" />
     <ClCompile Include="..\amxx_api.cpp" />
     <ClCompile Include="..\CstrikeHacks.cpp" />
+    <ClCompile Include="..\CstrikeHLTypeConversion.cpp" />
```
Comment (xPaw) [inline dlls/cstrike/cstrike/msvc12/cstrike.vcxproj]: Is this change supposed to be part of this pull request?

**Comment by KyleSanderson** [comment]
Not to jump in, but shouldn't this be a dynamic library?

**Comment by Arkshine** [comment]
I'm not sure to want to ship any extra DLLs apart the addon ones.

**Comment by dvander** [comment]
Sorry, I didn't mean to lose track of this (GitHub needs watch lists or something). It looks good, SourcePawn puts zlib in a third_party folder and doesn't call SetBuildFolder, but it doesn't really make a difference.

**Comment by dvander** [comment]
:ship:

**Comment by Arkshine** [comment]
Ahah don't worry, if it was important I would have spammed you!
I was not sure about SetBuildFolder, will keep this in mind if I touch again this file.

**Pull request merged**